src/auth/github.rs

```diff
@@ -0,0 +1,375 @@
+static GITHUB_CLIENT_ID_GLB: once_cell::sync::OnceCell<oauth2::ClientId> = once_cell::sync::OnceCell::new();
+static GITHUB_CLIENT_SECRET_GLB: once_cell::sync::OnceCell<oauth2::ClientSecret> = once_cell::sync::OnceCell::new();
+static AUTH_URL_GLB: once_cell::sync::OnceCell<oauth2::AuthUrl> = once_cell::sync::OnceCell::new();
+static TOKEN_URL_GLB: once_cell::sync::OnceCell<oauth2::TokenUrl> = once_cell::sync::OnceCell::new();
+
+use oauth2::TokenResponse;
+#[derive(Debug)]
+pub struct RepoObj {
+    pub repo_owner:String,
+    pub repo_title:String
+}
+pub fn index(req: actix_web::HttpRequest) -> actix_web::HttpResponse {
+    
+    let mut link="";
+    let mut link_title="";
+    match req.cookie("access_token"){
+        Some(val)=>{
+            //dbg!(val.value());
+            link="auth/logout/";
+            link_title="Logout";
+        }
+        None=>{
+            link="auth/login/";
+            link_title="Login";
+        }
+    }
+    let welcome_msg;
+    match req.cookie("user_login"){
+        Some(val)=>{
+            //dbg!(val.value());
+            welcome_msg=format!("{}{}","Hello ",val.value());
+        }
+        None=>{
+            welcome_msg=String::from("Welcome. Please first login: ");
+        }
+    }
+    let html = format!(
+        r#"<html>
+        <head><title>FDM</title></head>
+        <body>
+            {} <a href="/{}">{}</a>
+        </body>
+    </html>"#,
+    welcome_msg,
+        link,
+        link_title
+    );
+
+    actix_web::HttpResponse::Ok().body(html)
+}
+pub async fn login(req: actix_web::HttpRequest) -> actix_web::HttpResponse {
+     if GITHUB_CLIENT_ID_GLB.get().is_none(){
+        GITHUB_CLIENT_ID_GLB.set(oauth2::ClientId::new(
+            "77c964a9f6a7106a5a0e".to_string()
+        )).unwrap();
+     }  
+     if GITHUB_CLIENT_SECRET_GLB.get().is_none(){
+    GITHUB_CLIENT_SECRET_GLB.set(oauth2::ClientSecret::new(
+        "916d6cc2e912082f89891120b929680494467ba6".to_string()
```

Comment:
Don't check in secrets. Read them from environment variables.

Reply:
done